### PR TITLE
#80 Rooms joining/creation + searching for users frontend implementation

### LINKED ISSUE #80: GUI Room Join/Create + User Searching on Frontend

### PR DESCRIPTION
**Pull request by michaelKurowski**
**Reference to related ticket:**
 - resolves #80 

**Description of changes:**
 -  Joining room by its name
 -  Dialing room from connected rooms list
 - Searching users by username

### CODE REVIEW

**Comment by michaelKurowski** [comment]
TODO: tests

File: src/server/frontEnd/src/components/chatpage.js [hunk 1]
```diff
@@ -19,44 +19,89 @@ class ChatPage extends React.Component {
...
+		this.setState({roomToJoin: event.target.value})
+	}
+
+	handleUserToFindInput(event) {
+		const userToFind = event.target.value
+		this.setState({userToFind})
+		socket.users.emit(protocols.FIND, {queryPhrase: userToFind})
+	}
+
+	handleJoinEvent(data) {		if (data.username !== this.state.username) return
```
Comment (MarcinGrzeszczak) [inline src/server/frontEnd/src/components/chatpage.js]: fix spacing please

File: src/server/frontEnd/src/components/chatpage.js [hunk 2]
```diff
@@ -67,73 +112,125 @@ class ChatPage extends React.Component {
...
+						<span className='text-muted'>{new Date(msg.timestamp).toLocaleTimeString()}</span>
+					</p>
+				</li>
 		)
 	}
+
 	generateRooms() {
 		if(_.isEmpty(this.state.userRooms)) return
 		return this.state.userRooms.map(
 			(e, i) =>
```
Comment (MarcinGrzeszczak) [inline src/server/frontEnd/src/components/chatpage.js]: refactor args in map method please

File: src/server/frontEnd/src/components/chatpage.js [hunk 3]
```diff
@@ -67,73 +112,125 @@ class ChatPage extends React.Component {
...
+	printUsersInRoom() {
+		return this.state.usersInRoom.map(user => <li className='list-group-item' key={user}> {user} </li>)
+	}
+
+	createRoom(usernamesToInvite) {
+		socket.room.emit(protocols.CREATE, {invitedUsersIndexes: usernamesToInvite})
+	}
+
 	render() {
 		/* istanbul ignore next */
```
Comment (MarcinGrzeszczak) [inline src/server/frontEnd/src/components/chatpage.js]: remove istanbul command

**Pull request merged**